tap_universal_file/tap.py

```diff
@@ -299,14 +310,20 @@ def discover_streams(self) -> list[streams.FileStream]:
         Returns:
             A list of discovered streams.
         """
+        args = {}
         name = self.config["stream_name"]
         file_type = self.config["file_type"]
+        args["name"] = name
+        if self.config["schema_file"]:
+            schema_file = f"{SCHEMAS_DIR}/{self.config['schema_file']}"
+            args["schema"] = schema_file
+
         if file_type == "delimited":
-            return [streams.DelimitedStream(self, name=name)]
+            return [streams.DelimitedStream(self, **args)]
         if file_type == "jsonl":
-            return [streams.JSONLStream(self, name=name)]
+            return [streams.JSONLStream(self, **args)]
         if file_type == "avro":
-            return [streams.AvroStream(self, name=name)]
+            return [streams.AvroStream(self, **args)]
```

Comment:
We should add a sampling strategy for each of our file types of something like "manual". The manual sampling strategy would allow you to do a few things

1. Point to a schema file directly as you pointed to here
2. Pass in a schema via the Normal method for Singer which is via the Catalog (ie bypass Discovery)

This is nice as it'd allow folks to specify exactly what they want the schema to look like while also respecting Singer defaults. I'm not so sure we really need to support a schema file but I can see where that might be useful!